service/delete_app_consumer.go

```diff
@@ -0,0 +1,67 @@
+package service
+
+import (
+	"context"
+
+	"github.com/getAlby/hub/events"
+	"github.com/getAlby/hub/logger"
+	"github.com/nbd-wtf/go-nostr"
+)
+
+type deleteAppConsumer struct {
+	events.EventSubscriber
+	walletPubkey      string
+	relay             *nostr.Relay
+	nostrSubscription *nostr.Subscription
+	svc               *service
+}
+
+// When an app is deleted, unsubscribe from events for that app on the relay
+// and publish a deletion event for that app's info event
+func (s *deleteAppConsumer) ConsumeEvent(ctx context.Context, event *events.Event, globalProperties map[string]interface{}) {
+	if event.Event != "app_deleted" {
+		return
+	}
+	properties, ok := event.Properties.(map[string]interface{})
+	if !ok {
+		logger.Logger.WithField("event", event).Error("Failed to cast event.Properties to map")
+		return
+	}
+	id, ok := properties["id"].(uint)
+	if !ok {
+		logger.Logger.WithField("event", event).Error("missing id in properties event")
+		return
+	}
+
+	walletPrivKey, err := s.svc.keys.GetAppWalletKey(id)
+	if err != nil {
+		logger.Logger.WithError(err).WithField("id", id).Error("Failed to calculate app wallet priv key")
+		return
+	}
+	walletPubKey, err := nostr.GetPublicKey(walletPrivKey)
+	if err != nil {
+		logger.Logger.WithError(err).WithField("id", id).Error("Failed to calculate app wallet pub key")
+		return
+	}
+	// Note: for legacy apps this check will always return false as the wallet pubkey
+	// generated by the id will not match the master key which is used for all legacy apps
+	if s.walletPubkey == walletPubKey {
```

Comment:
Isn't this listener ConsumeEvent called for each Subscriber added, meaning as many as each app connection wallet? And we want to only do something for the specific one matching the `walletPubkey` we saved in `deleteAppConsumer` struct rigth?

Reply:
I am not sure it's worth it to add this extra code. The old info event doesn't really do any harm, does it? (there will be others - e.g. if users abandon old hubs there will be many info events for apps that are no longer used and were never deleted)

Reply:
> And we want to only do something for the specific one matching the walletPubkey we saved in deleteAppConsumer struct right?

Whoops sorry, it should be `s.walletPubKey == walletPubkey || (s.walletPubKey == s.svc.keys.GetNostrPublicKey() && legacyAppCount == 0)`

But yeah we can leave the legacy info event now as it does no harm 👍

Reply:
One concern though, can we just pass in the `app.WalletPubkey` in the eventPublisher instead of deriving the key for every app when some app is deleted? So we can only do key derivation once per delete after the `s.walletPubkey == properties["walletPubkey"]` check

Reply:
> > And we want to only do something for the specific one matching the walletPubkey we saved in deleteAppConsumer struct right?
> 
> Whoops sorry, it should be `s.walletPubKey == walletPubkey || (s.walletPubKey == s.svc.keys.GetNostrPublicKey() && legacyAppCount == 0)`
> 
> But yeah we can leave the legacy info event now as it does no harm 👍

We would have to get an up-to-date `legacyAppCount` from DB to check that it was the last legacy app being deleted. So I think the legacy event info can stay undeleted, no harm.

Reply:
> One concern though, can we just pass in the `app.WalletPubkey` in the eventPublisher instead of deriving the key for every app when some app is deleted? So we can only do key derivation once per delete after the `s.walletPubkey == properties["walletPubkey"]` check

I was doing that originally, but @rolznz suggested to be consistent and pass `app.ID` and derive there.
The derivation impact on app deletion for all existing subscribers/apps shouldn't be much of an issue, unless you have 1000's of apps, right?

Reply:
Hmm... good point. Couldn't the delete app consumer store the app ID? then it can do a simple check before any deriving.

Reply:
Yeah we can do that if required later 👍